src/meta/src/hummock/manager/mod.rs

```diff
@@ -777,12 +777,24 @@ where
         let can_trivial_move = matches!(selector.task_type(), compact_task::TaskType::Dynamic);
 
         let mut stats = LocalSelectorStatistic::default();
+        let table_id_to_option = self
```

Comment:
Since `CatalogManager` may hold the lock for a long time when it runs a DDL job, I think we shall call `get_table_options` before getting guard of `compaction` lock.

Reply:
The context here is that we need to get the current_version before we can get the corresponding table_id, although table_id_option can be lagged for ttl

What do you think @zwang28

Reply:
Alternatively we can get all table options before compaction lock.